[Focus Mode] Images and content for the breakpoint article (#2931)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/javascript/breakpoints.md

1. Select **Add conditional breakpoint**.  A dialog is displayed underneath the line of code.

   ![The dialog that appears when setting a conditional breakpoint](./breakpoints-images/conditional-breakpoint-dialog.png)

1. Enter your condition in the dialog.

1. Press **Enter** to activate the breakpoint.  A red icon appears next to the line number column:

   ![A conditional line-of-code breakpoint](./breakpoints-images/conditional-breakpoint.png)

#### Manage line-of-code breakpoints

Use the **Breakpoints** pane to disable or remove line-of-code breakpoints from a single location.

![The Breakpoints panel](./breakpoints-images/breakpoints-16-33.png)

*  Select the checkbox next to an entry to disable that breakpoint.

*  Right-click an entry to remove that breakpoint.

*  Right-click anywhere in the **Breakpoints** pane to remove all breakpoints.

> To disable all breakpoints, click **Deactive breakpoints** (![The Deactivate breakpoints icon](./breakpoints-images/deactivate-breakpoints-icon.png)) button:

Reviewer: ```suggestion
To disable all breakpoints, click the **Deactive breakpoints** (![The Deactivate breakpoints icon](./breakpoints-images/deactivate-breakpoints-icon.png)) button:
```

Reviewer: done

1. Go the element on which you want to set the breakpoint.

> 1. Right-click the element, click **Break on**, and then click **subtree modifications**, **attribute modifications**, or **node removal**.

Reviewer: ```suggestion
1. Right-click the element, click **Break on**, and then click **subtree modifications**, **attribute modifications**, or **node removal**:
```
rendered:
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/breakpoints?branch=pr-en-us-2931#dom-change-breakpoints

Reviewer: done

<!-- ====================================================================== -->
## XHR/fetch breakpoints

Use an XHR/fetch breakpoint when you want to break when an [XmlHttpRequest](https://developer.mozilla.org/docs/Web/API/XMLHttpRequest) (XHR) or [Fetch](https://developer.mozilla.org/docs/Web/API/Fetch_API) request occurs. DevTools pauses on the line of code where the XHR or Fetch request occurs.

One example of when this is helpful is when your webpage is requesting an incorrect URL, and you want to quickly find the XHR or Fetch source code that is causing the incorrect request.

To set an XHR breakpoint:

1. Select the **Sources** tool.

1. Expand the **XHR/fetch Breakpoints** panel.

1. Click **Add breakpoint**.

> 1. Enter all or part of the URL which you want to break on.  DevTools pauses when the value you enter is present anywhere in an XHR or Fetch request URL.

Reviewer: ```suggestion
1. Enter all or part of the URL that you want to break on.  DevTools pauses when the value you enter is present anywhere in an XHR or Fetch request URL.
```
according to style guide
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/t/that-vs-which

Reviewer: done

1. Expand the **Event Listener Breakpoints** panel.  DevTools shows a list of event categories, such as **Animation**.

> 1. Check one of these categories to pause whenever any event from that category is fired, or expand the category and check a specific event:

Reviewer: ```suggestion
1. Select one of these categories to pause whenever any event from that category is fired.  Or, expand the category, and then select a specific event:
```
rendered:
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/breakpoints?branch=pr-en-us-2931#event-listener-breakpoints

Reviewer: done locally

1. Select the **Sources** tool.

1. In the **Breakpoints** panel, select **Pause on uncaught exceptions** to pause when the code throws a JavaScript exception.

> 1. **Optional:** Select **Pause on caught exceptions** if you also want to pause on caught exceptions, for example when an exception is caught by a `try/catch` block.

Reviewer: ```suggestion
1. **Optional:** Select **Pause on caught exceptions** if you also want to pause on caught exceptions - for example, when an exception is caught by a `try/catch` block:
```

Reviewer: rewrote the two items as:

1. To pause when the code throws a JavaScript exception, in the **Breakpoints** pane, select the **Pause on uncaught exceptions** checkbox.

1. To pause on caught exceptions, such as when an exception is caught by a `try/catch` block, select the **Pause on caught exceptions** checkbox:

<!-- ====================================================================== -->
## Function breakpoints

Run the `debug(function)` method, where `function` is the JavaScript function that you want to debug, when you want to pause whenever a specific function is run.  You can insert `debug()` into your code (such as when using a `console.log()` statement), or run the method from the DevTools **Console** tool.

`debug()` is equivalent to setting a [line-of-code breakpoint](#line-of-code-breakpoints) on the first line of the function.

```javascript
function sum(a, b) {
  let result = a + b; // DevTools will pause before running this line.
  return result;
}

// Call the debug method by passing a reference to the function object,
// not its name as a string.
debug(sum);

sum();
```

#### Make sure the target function is in scope

DevTools throws a `ReferenceError` if the function you want to debug isn't in scope.

```javascript
(function () {
  function foo() {
    console.log('foo');
  }

  function bar() {
    console.log('bar');
  }

  // Here, calling debug(bar) works because
  // bar is defined in the current scope.
  debug(bar);

  bar();
})();

> // Here, calling debug(foo) won't work because foo

Reviewer: ```suggestion
// Here, calling debug(foo) won't work, because foo
```

Reviewer: done
